test/e2e/specs/editor/various/inserting-blocks.spec.js

```diff
@@ -386,6 +386,323 @@ test.describe( 'Inserting blocks (@firefox, @webkit)', () => {
+			.click();
+		await page
+			.getByRole( 'listbox', { name: 'Text' } )
+			.getByRole( 'option', { name: 'Heading' } )
+			.hover();
+
+		await expect( insertingBlocksUtils.indicator ).toBeVisible();
+
+		const paragraphBoundingBox = await paragraphBlock.boundingBox();
+		const indicatorBoundingBox =
+			await insertingBlocksUtils.indicator.boundingBox();
+
+		// Expect the indicator to be below the selected block.
+		expect( indicatorBoundingBox.y ).toBeGreaterThan(
+			paragraphBoundingBox.y
+		);
+	} );
+
+	// Check for regression of https://github.com/WordPress/gutenberg/issues/24403.
+	test( 'inserts a block in proper place after having clicked `Browse All` from block appender', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		await admin.createNewPost();
+		await editor.insertBlock( { name: 'core/group' } );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Paragraph after group' },
+		} );
+
+		await editor.canvas
+			.getByRole( 'button', {
+				name: 'Group: Gather blocks in a container.',
+			} )
+			.click();
+		await editor.canvas
+			.getByRole( 'button', {
+				name: 'Add block',
+			} )
+			.click();
+		await page.getByRole( 'button', { name: 'Browse All' } ).click();
+		await page
+			.getByRole( 'listbox', { name: 'Text' } )
+			.getByRole( 'option', { name: 'Paragraph' } )
+			.click();
+		await editor.canvas
+			.getByRole( 'document', { name: 'Empty block' } )
+			.fill( 'Paragraph inside group' );
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/group',
+				innerBlocks: [
+					{
+						name: 'core/paragraph',
+						attributes: { content: 'Paragraph inside group' },
+					},
+				],
+			},
+			{
+				name: 'core/paragraph',
+				attributes: { content: 'Paragraph after group' },
+			},
+		] );
+	} );
+
+	test( 'passes the search value in the main inserter when clicking `Browse all`', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		await admin.createNewPost();
+		await editor.insertBlock( { name: 'core/group' } );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Paragraph after group' },
+		} );
+
+		await editor.canvas
+			.getByRole( 'button', {
+				name: 'Group: Gather blocks in a container.',
+			} )
+			.click();
+		await editor.canvas
+			.getByRole( 'button', {
+				name: 'Add block',
+			} )
+			.click();
+
+		const searchBox = page.getByRole( 'searchbox', {
+			name: 'Search for blocks and patterns',
+		} );
+
+		await searchBox.fill( 'Verse' );
+		await page.getByRole( 'button', { name: 'Browse All' } ).click();
+
+		await expect( searchBox ).toHaveValue( 'Verse' );
+		await expect(
+			page.getByRole( 'listbox', { name: 'Blocks' } )
+		).toHaveCount( 1 );
+	} );
+
+	// Check for regression of https://github.com/WordPress/gutenberg/issues/27586.
+	test( 'can close the main inserter after inserting a single-use block, like the More block', async ( {
```

Comment:
I have updated the test name here to reflect the intention better. Clicking on any black option in the inserter doesn't close it. The `insertBlock` help utility provided the original test behavior.